deep_field_metadetect/gaussmom/gaussmom.py

```diff
@@ -0,0 +1,417 @@
+    -------
+    result array with basic sums as well as summary statistics
+    such as e1,e2,T,s2n etc.
+    """
+
+    sums, sums_cov, wsum, _ = get_weighted_sums(
+        obs,
+        T=T,
+        maxrad=maxrad,
+    )
+    return get_weighted_moments_stats(obs, sums, sums_cov, wsum)
+
+
+# @jax.jit
+def get_weighted_sums(gaussmom_obs, T, maxrad):
+    """
+    Compute weighted moment sums and their covariance for a 2D Gaussian model.
+
+    This function evaluates the weighted image moments within a circular region
+    defined by `maxrad`, using a Gaussian weight function with size parameter `T`.
+    It returns the weighted sums, the covariance of the sums, total weight,
+    and the normalization factor for the weight function.
+
+    Parameters
+    ----------
+    gaussmom_obs : object
+        An object containing observation data with the following attributes:
+            - u, v: pixel coordinates
+            - image: observed image data
+            - pixelwise_wgt: inverse variance for each pixel
+            - area: pixel area
+    T : float
+        Size (trace) of the weighting Gaussian used for computing moments.
+    maxrad : float
+        Maximum radius (in u, v coordinates) for including pixels in the calculation.
+
+    Returns
+    -------
+    sums : jax.numpy.ndarray
+        Array of shape (6,) containing the weighted image moment sums:
+            [v, u, u^2 - v^2, 2uv, u^2 + v^2, 1]
+    sums_cov : jax.numpy.ndarray
+        Covariance matrix of shape (6, 6) for the weighted sums.
+    wsum : float
+        Total sum of the weights applied to the image.
+    wt_norm : float
+        Normalization factor for the Gaussian weight function.
+    """
+    vcen = 0
+    ucen = 0
+
+    mompars = [0, 0, T / 2, 0, T / 2]
+
+    vmod = gaussmom_obs.v - vcen
+    umod = gaussmom_obs.u - ucen
+
+    var = 1.0 / (gaussmom_obs.pixelwise_wgt)
+    rad2 = umod * umod + vmod * vmod
+
+    circle_mask = jnp.where(jnp.sqrt(rad2) <= maxrad, 1, 0)
+
+    wt_noimage, wt_norm = _eval_gauss2d(
+        mompars, gaussmom_obs.u, gaussmom_obs.v, gaussmom_obs.area
+    )
+
+    wt_noimage = wt_noimage / (wt_norm * gaussmom_obs.area)
+    wt_noimage = circle_mask * wt_noimage
+
+    wdata = wt_noimage * gaussmom_obs.image
+
+    # print(jnp.sum(umod * umod + vmod * vmod))
+
+    F = jnp.stack(
+        [
+            gaussmom_obs.v,
+            gaussmom_obs.u,
+            umod * umod - vmod * vmod,
+            2 * vmod * umod,
+            rad2,
+            jnp.ones_like(gaussmom_obs.v),
+        ]
+    )
+
+    wsum = jnp.sum(wt_noimage)
+    # res["npix"] = jnp.sum(circle_mask)
+
+    sums = jnp.sum(F * wdata, axis=[1, 2])
+
+    sums_cov = jnp.zeros((6, 6))
+
+    for i in range(6):
+        for j in range(6):
+            sums_cov = sums_cov.at[i, j].set(jnp.sum(wt_noimage**2 * var * F[i] * F[j]))
+
+    return sums, sums_cov, wsum, wt_norm
+
+
+# @jax.jit
+def get_weighted_moments_stats(obs, sums, sums_cov, sums_norm=None):
```

Comment:
We'd like this to be able to be jitted. The current control flow will not enable that.

Reply:
done!